### PR TITLE
Add response status class counters

### PR DESCRIPTION
**Pull request by charleso**
/cc @markhibberd @tmcgilchrist

### CODE REVIEW

File: Network/Wai/Metrics.hs [hunk 1]
```diff
@@ -86,6 +97,12 @@ metrics waiMetrics app req respond = do
     where respond' :: UTCTime -> Response -> IO ResponseReceived
           respond' start res = do
             when (statusIsServerError $ responseStatus res) (Counter.inc (serverErrorCounter waiMetrics))
```
Comment (charleso) [inline Network/Wai/Metrics.hs]: Arguably we could remove this now

File: README.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 A [WAI](https://hackage.haskell.org/package/wai) middleware to collect the following [EKG](https://ocharles.org.uk/blog/posts/2012-12-11-24-day-of-hackage-ekg.html) metrics from compatible web servers:
 - number of requests (counter `wai.request_count`)
 - number of server errors (counter `wai.server_error_count`)
+- number of response by status code, broken down class (counter `wait.response_status_xxx`, eg `wait.response_status_2xx`)
 - latency distribution (distribution `wai.latency_distribution`)
```
Comment (tmcgilchrist) [inline README.md]: s/wait/wai

**Comment by tmcgilchrist** [comment]
Cool :+1:

File: Network/Wai/Metrics.hs (hunk 1 above)
Comment (tmcgilchrist) [inline Network/Wai/Metrics.hs]: If as @charleso indicated this is just counting the 500 errors under a different name. I think we can just go with the explicit 500 error code below.

**Pull request merged**